addons/account/models/account_account.py

```diff
@@ -506,12 +506,19 @@ def default_get(self, default_fields):
         return super(AccountAccount, contextual_self).default_get(default_fields)
 
     @api.model
-    def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None, limit=None):
+    def _get_most_used_accounts_for_partner(self, company_id, partner_id, move_type, join='left join', limit=None):
         """
-        Returns the accounts ordered from most used to least used for a given partner
-        and filtered according to the move type.
+        Returns the accounts ordered from most used to least used for a given partner and filtered according to the move type
+        :param company_id: the company id
+        :param partner_id: the partner id for which we want to retrieve the most frequent accounts
+        :param move_type: the type of the move to know which type of accounts to retrieve
+        :param join: join method. Either 'inner join' (filters out accounts never used for the partner) or 'left join' (no filter)
+        :param limit: the maximum number of accounts to retrieve
+        :returns: List of account ids, ordered by frequency (from most to least frequent)
         """
-        limit = "LIMIT 1" if limit is not None else ""
+        assert join in ('inner join', 'left join')
+        assert limit is None or (isinstance(limit, int) and limit > 0)
+        limit = f"LIMIT {limit}" if limit else ""
```

Comment:
No need to assert, you can just format it
```suggestion
        limit = f"LIMIT {limit:d}" if limit else ""
```